src/main/java/xbot/common/controls/sensors/XGamepad.java

```diff
@@ -0,0 +1,70 @@
+package xbot.common.controls.sensors;
+
+import xbot.common.injection.wpi_factories.CommonLibFactory;
+import xbot.common.logging.RobotAssertionManager;
+import xbot.common.math.XYPair;
+
+public abstract class XGamepad extends XJoystick {
+
+	// All of the "Left Joystick" inversions are kept track of by the base XJoystick class.
+	// We just need to redirect any calls to the base class.
+	private boolean xRightInverted = false;
+    private boolean yRightInverted = false;
+
+	public XGamepad(int port, CommonLibFactory clf, RobotAssertionManager assertionManager, int numButtons) {
+		super(port, clf, assertionManager, numButtons);
+	}
+	
+	public XYPair getRightVector() {
+        return new XYPair(getRightStickX(), getRightStickY());
+    }
+
+    public XYPair getLeftVector() {
+        return this.getVector();
+    }
+    
+    public double getRightStickX(){
+        return this.getRawAxis(4) * (getRightStickXInversion() ? -1 : 1);
```

Comment:
My recollection is that these axes are inconsistent between Windows versions/drivers; where did these numbers come from?

Reply:
I should actually be more specific - this is a FTCGamepad implementation (or whatever the proper brand name of the FTC gamepads is). Those have been very consistent. I believe it's the Xbox controllers that have been inconsistent in the past.